src/vault/client.clj

```diff
@@ -25,33 +32,80 @@
 
 ;; ## HTTP API Client
 
+(defn- check-path!
+  "Validates that the given path."
+  [path]
+  (when-not (string? path)
+    (throw (IllegalArgumentException.
+             (str "Secret path must be a string, got: " (pr-str path))))))
+
+
+(defn- check-auth!
+  "Validates that the client is authenticated."
+  [token-ref]
+  (when-not @token-ref
+    (throw (IllegalStateException.
+             "Cannot read path with unauthenticated client."))))
+
+
+(defn- authenticate-token!
+  [token-ref token]
+  (when-not (string? token)
+    (throw (IllegalArgumentException. "Token credential must be a string")))
+  ; TODO: test auth?
```

Comment:
From in-person discussion this is not valid, so they'll just get an error later.